src/SlackGhost.ts

```diff
@@ -312,6 +316,29 @@ export class SlackGhost {
         return Slackdown.parse(body);
     }
 
+    public async sendInThread(roomId: string, text: string, slackRoomId: string,
+        slackEventTs: string, replyEvent: IMatrixReplyEvent): Promise<void> {
+        const fallbackHtml = this.getFallbackHtml(roomId, replyEvent);
+        const fallbackText = this.getFallbackText(replyEvent);
+
+        const content = {
+            "m.relates_to": {
+                "rel_type": "io.element.thread",
+                // If the reply event is part of a thread, continue the thread.
+                // Otherwise, attach a thread to the reply event.
+                "event_id": replyEvent.content["m.relates_to"]?.event_id ?? replyEvent.event_id,
+                "m.in_reply_to": {
+                    event_id: replyEvent.event_id,
+                },
+            },
+            "msgtype": "m.text", // for those who just want to send the reply as-is
+            "body": `${fallbackText}\n\n${this.prepareBody(text)}`,
```

Comment:
So I'm not sure if you still need a fallback or not, looking at what Element does:

![image](https://user-images.githubusercontent.com/2072976/157060860-7b2e136b-b342-4340-bdfd-cfa2c8a7fa1f.png)

Reply:
Element does not provide this fallback.
I guess if a client or bridge (e.g. IRC) wants to display the relation, it up to that cleint's or bridge's implementation to provide an adequate representation.